core/src/main/scala/io/finch/package.scala

```diff
@@ -23,4 +26,34 @@ package object finch extends Outputs with ValidationRules {
     case object MultipleItems extends RequestItem("request")
   }
 
+  implicit class CompiledOps[F[_]](compiled: Endpoint.Compiled[F]) {
```

Comment:
Let's chat here what others ideas we might have around this?

Just a simple class `ToService` could do it:

```scala
case class ToService(k: Endpoint.Compiled[IO]) extends Service[Req, Rep] { ... }

// Usage:
val k: Endpoint.Compiled[IO] = Endpoint[IO].const("foo").compile
val s = ToService(k)
```

By the way, what do you think about adding `compile` on `Endpoint`?

Reply:
Personally, I would prefer to push users toward using `Bootstrap`

Reply:
I agree. But we do have `toService` on an `Endpoint`. I think it wouldn't hurt to add compile as a shortcut. I don't think it's confusing whether or not people should use `Bootstrap` (no semantic changes to that).

Reply:
How about `Endpoint.run` or `Endpoint.service` as the function to convert compiled endpoint into `Service`?

Reply:
Here is a terrible idea. The only type we can hijack in `Endpoint.Compiled[F]` is `Trace`. Turns out if we put our syntax extension under `Trace` companion object, the compiler is able to find it:

```scala
scala> import cats.data.Kleisli, cats.effect.IO
import cats.data.Kleisli
import cats.effect.IO

scala> trait Trace; object Trace { implicit class KlesliSyntax(k: Kleisli[IO, Int, (Trace, Either[Throwable, String])]) { def toService: Unit = () } }
       trait Trace; object Trace { implicit class KlesliSyntax(k: Kleisli[IO, Int, (Trace, Either[Throwable, String])]) { def toService: Unit = () } }
                                                                                                                              ^
defined trait Trace
defined object Trace

scala> val k = Kleisli.pure[IO, Int, (Trace, Either[Throwable, String])](new Trace{} -> Right("foo"))
k: cats.data.Kleisli[cats.effect.IO,Int,(Trace, Either[Throwable,String])] = Kleisli(cats.data.KleisliFunctions$$Lambda$4443/1717142056@5206bb69)

scala> k.toService
()                          ^
```

Reply:
While it kinda hides it from the users radar, I think it's quite ugly from the perspective of internal maintance and code organization 😄

Reply:
I agree. I still like `toService` as a name just because it's something we've been using for quite some time. Although I'm out of ideas where to define it.